pkg/plugin/sdk/deployment.go

```diff
@@ -206,7 +206,33 @@ func (s *DeploymentPluginServiceServer[Config, DeployTargetConfig]) ExecuteStage
 		stageID:       request.GetInput().GetStage().GetId(),
 		logPersister:  lp,
 	}
-	return executeStage(ctx, s.base, &s.config, nil, client, request, s.logger) // TODO: pass the deployTargets
+
+	dtNames, err := request.GetInput().GetDeployment().GetDeployTargets(s.commonFields.config.Name)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "failed to get deploy targets for plugin %s: %v", s.commonFields.config.Name, err)
```

Comment:
[nits]
[nit]
I wanna clarify which one we mention,
- the deployTraget was not found in a piped config
- the deployTarget was not set in the request(deployment)

```suggestion
		return nil, status.Errorf(codes.Internal, "failed to get deploy targets from the deployment for plugin %s: %v", s.commonFields.config.Name, err)
```

Reply:
Fixed on [fbb133e](https://github.com/pipe-cd/pipecd/pull/5616/commits/fbb133e012a69c6715bbced9af4a1bd07174d611)